frontend/lib/resolution/InitResolver.h

```diff
@@ -129,6 +129,9 @@ class InitResolver {
 
 public:
 
+  //initialization points to guide handling `=` operators 
+  std::set<const uast::AstNode*> initPoints;
```

Comment:
I think it would be better to add a method on InitResolver that returns true if a given AstNode is field initialization.